Sources/ParseSwift/Authentication/3rd Party/ParseTwitter.swift

```diff
@@ -0,0 +1,267 @@
+ For information on acquiring Twitter sign-in credentials to use with `ParseTwitter`, refer to [Twitter's Documentation](https://developer.twitter.com/en/docs/authentication/guides/log-in-with-twitter.
+ */
+public struct ParseTwitter<AuthenticatedUser: ParseUser>: ParseAuthentication {
+
+    /// Authentication keys required for Twitter authentication.
+    enum AuthenticationKeys: String, Codable {
+        case id // swiftlint:disable:this identifier_name
+        case token
+        case tokenSecret
+
+        /// Properly makes an authData dictionary with the required keys.
+        /// - parameter twitterId: Required id for the user.
+        /// - parameter authToken: Required Twitter authToken obtained from Twitter SDK for the user.
+        /// - parameter authTokenSecret: Required Twitter authSecretToken obtained from Twitter SDK for the user.
+        /// - returns: Required authData dictionary.
+        func makeDictionary(twitterId: String,
+                            authToken: String,
+                            authTokenSecret: String) -> [String: String]? {
+
+            return [AuthenticationKeys.id.rawValue: twitterId,
+             AuthenticationKeys.token.rawValue: authToken,
+             AuthenticationKeys.tokenSecret.rawValue: authTokenSecret]
+        }
+
+        /// Verifies all mandatory keys are in authData.
+        /// - parameter authData: Dictionary containing key/values.
+        /// - returns: `true` if all the mandatory keys are present, `false` otherwise.
+        func verifyMandatoryKeys(authData: [String: String]?) -> Bool {
+            guard let authData = authData,
+                  authData[AuthenticationKeys.id.rawValue] != nil,
+                  authData[AuthenticationKeys.token.rawValue] != nil,
+                  authData[AuthenticationKeys.tokenSecret.rawValue] != nil else {
+                return false
+            }
+            return true
+        }
+    }
+    public static var __type: String { // swiftlint:disable:this identifier_name
+        "twitter"
+    }
+    public init() { }
+}
+
+// MARK: Login
+public extension ParseTwitter {
+    /**
+     Login a `ParseUser` *asynchronously* using Twitter authentication.
+     - parameter twitterId: The `Twitter userId` from `Twitter session`.
+     - parameter username: The `Twitter username` from `Twitter session`.
+     - parameter authToken: The Twitter `authToken` obtained from Twitter SDK for the user.
+     - parameter authTokenSecret: The Twitter `authSecretToken` obtained from Twitter SDK for the user
+     - parameter options: A set of header options sent to the server. Defaults to an empty set.
+     - parameter callbackQueue: The queue to return to after completion. Default value of .main.
+     - parameter completion: The block to execute.
+     */
+
+    func login(withTwitterId: String,
+               username: String,
+               authToken: String,
```

Comment:
Need to modify the methods to take the keys I mentioned above, also, remove `with` from `withTwitterId`. Be sure to do this for all of the sync, async, and publisher methods

Reply:
For twitter, your method will look like:

```swift
func login(twitterId: String,
                screenName: String,
                authToken: String,
                authTokenSecret: String,
                consumerKey: String,
                consumerSecret: String,
                options: API.Options = [],
                callbackQueue: DispatchQueue = .main,
                completion: @escaping (Result<AuthenticatedUser, ParseError>) -> Void) {
```